src/common/database.c

```diff
@@ -2729,17 +2746,24 @@ int _get_pragma_val(const struct dt_database_t *db, const char* pragma)
 void dt_database_maybe_maintenance(const struct dt_database_t *db, const gboolean has_gui, const gboolean closing_time)
 {
   gboolean check_for_maintenance = FALSE;
+  gboolean force_maintenance = FALSE;
 
   char *config = dt_conf_get_string("database/maintenance_check");
 
```

Comment:
I suppose this here would work:

`const gboolean force_maintenance = g_str_has_suffix (config, "(don't ask)");`